src/Adapter/Product/ProductSupplierUpdater.php

```diff
@@ -0,0 +1,186 @@
+
+    /**
+     * @param ProductId $productId
+     * @param SupplierId $defaultSupplierId
+     * @param ProductSupplier[] $productSuppliers
+     *
+     * @throws CannotUpdateProductSupplierException
+     * @throws CoreException
+     * @throws ProductSupplierNotFoundException
+     */
+    public function setProductSuppliers(ProductId $productId, SupplierId $defaultSupplierId, array $productSuppliers): void
+    {
+        $deletableProductSupplierIds = $this->getDeletableProductSupplierIds($productId->getValue(), $productSuppliers);
+
+        foreach ($productSuppliers as $productSupplier) {
+            if ($productSupplier->id) {
+                $this->productSupplierRepository->update($productSupplier);
+            } else {
+                $this->productSupplierRepository->add($productSupplier);
+            }
+        }
+
+        $this->productSupplierRepository->bulkDelete($deletableProductSupplierIds);
+        $this->updateDefaultSupplier($productId, $defaultSupplierId);
+    }
+
+    /**
+     * @param Product $product
+     */
+    public function resetDefaultSupplier(Product $product): void
+    {
+        $product->supplier_reference = '';
+        $product->wholesale_price = 0;
+        $product->id_supplier = 0;
+
+        $this->productRepository->partialUpdate(
+            $product,
+            ['supplier_reference' => true, 'wholesale_price' => true, 'id_supplier' => true],
+            CannotUpdateProductException::FAILED_UPDATE_DEFAULT_SUPPLIER
+        );
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param SupplierId $supplierId
+     */
+    public function updateDefaultSupplier(ProductId $productId, SupplierId $supplierId): void
+    {
+        $product = $this->productRepository->get($productId);
+        $supplierIdValue = $supplierId->getValue();
+        $productIdValue = (int) $product->id;
+
+        if ($product->hasCombinations()) {
+            $this->resetDefaultSupplier($product);
+
+            return;
+        }
+
+        if ((int) $product->id_supplier === $supplierIdValue) {
+            return;
+        }
+
+        $this->supplierRepository->assertSupplierExists($supplierId);
+        $productSupplierId = (int) ProductSupplier::getIdByProductAndSupplier($productIdValue, 0, $supplierIdValue);
+
+        if (!$productSupplierId) {
+            throw new ProductSupplierNotFoundException(sprintf(
+                'Supplier #%d is not associated with product #%d', $supplierIdValue, $productIdValue
+            ));
+        }
+
+        $product->supplier_reference = ProductSupplier::getProductSupplierReference($productIdValue, 0, $supplierIdValue);
+        $product->wholesale_price = ProductSupplier::getProductSupplierPrice($productIdValue, 0, $supplierIdValue);
+        $product->id_supplier = $supplierIdValue;
+
+        $this->productRepository->partialUpdate(
+            $product,
+            ['supplier_reference' => true, 'wholesale_price' => true, 'id_supplier' => true],
```

Comment:
Sorry I didn't realize in previous PR that the partial update behaved this way now (although you warned me but I didn't get it ^^) So we don't use an array with the values to fill but only the list of updated fields
I'm fine with it but couldn't it be made even easier by using a simple array? Like this:
```suggestion
            ['supplier_reference', 'wholesale_price', 'id_supplier'],
```

Or is it because of the localized fields?

Reply:
yes. Localized fields might introduce a problem :thinking: , but I think we can still get rid of those bools